publish_pipeline/generate_high_level_metadata/generate_license.py

```diff
@@ -0,0 +1,94 @@
+# pylint: disable=line-too-long
+"""Process environmental sensor data files"""
+
+import datetime
+import pathlib
+import tempfile
+import uuid
+
+import azure.storage.blob as azureblob
+import psycopg2
+import pyfairdatatools
+
+import config
+
+
+def pipeline():
+    """Reads the database for the dataset and generates a license.txt file in the metadata folder."""
+
+    license_metadata = {}
+
+    conn = psycopg2.connect(
+        host=config.FAIRHUB_DATABASE_HOST,
+        database=config.FAIRHUB_DATABASE_NAME,
+        user=config.FAIRHUB_DATABASE_USER,
+        password=config.FAIRHUB_DATABASE_PASSWORD,
+        port=config.FAIRHUB_DATABASE_PORT,
+    )
+
+    cur = conn.cursor()
+
+    study_id = "c588f59c-cacb-4e52-99dd-95b37dcbfd5c"
+    dataset_id = "af4be921-e507-41a9-9328-4cbb4b7dca1c"
+
+    cur.execute(
+        "SELECT * FROM dataset WHERE id = %s AND study_id = %s",
+        (dataset_id, study_id),
+    )
+
+    dataset = cur.fetchone()
+
+    if dataset is None:
+        return "Dataset not found"
+
+    license_text = ""
+
+    cur.execute(
+        "SELECT license_text FROM dataset_rights WHERE dataset_id = %s",
+        (dataset_id,),
+    )
+
+    dataset_other = cur.fetchone()
+    # license_text = dataset_other.join(",")
+    license_text = dataset_other[0]
+    license_metadata = license_text
+
+    conn.close()
+
+    # Create a temporary folder on the local machine
+    temp_folder_path = tempfile.mkdtemp()
+
+    temp_file_path = pathlib.Path(temp_folder_path, "license.txt")
+    pyfairdatatools.generate.generate_license_file(
+        data=license_metadata, file_path=temp_file_path, file_type="txt"
+    )
```

Comment:
This is not correct. I don't think the library can't be used in this instance(it takes an identifier, not text). You have to create the license manually.

Reply:
In that case in which circumstances can we use library? Perhaps we need to add license identifier or has to modify library to accept text instead of id?

Reply:
The library is for general use. I think our situation doesn't apply in this instance. You can create the license similar to how we would create the changelog in this instance